getting-started/data-node-authentication.md

```diff
@@ -11,90 +11,181 @@ and the access node need to have authentication set up for those
+```
+
+>:TIP: A quick way to edit the configuration file is by using Perl:
+>
+> ```
+> perl -pi -e "s/^#?(password_encryption) *= *\S+/\1 = 'scram-sha-256'/" \
+>    postgresql.conf
+> ```
+>
+> This will rename the original configuration file to
+> `postgresql.conf.orig` and set `password_encryption` in
+> `postgresql.conf`.
+
+### Enable password authentication
+
+The default name of the HBA file is `pg_hba.conf`, so check the
+configuration file (normally `postgresql.conf` in the data directory
+of the data node server) and make sure that the `hba_file` option is
+commented out. That part of the file should look like this:
+
+```
+#hba_file = 'ConfigDir/pg_hba.conf' # host-based authentication file
+                                    # (change requires restart)
+```
+
+After that, you should add a line to the `pg_hba.conf` file giving
+`new_user` in the `example.com` domain access to all databases on the
+data node and is identified using SCRAM SHA-256.
+
+```
+# IPv4 local connections:
+host    all             new_user        .example.com            password
+```
+
+Details of [the `pg_hba.conf` format can be found in the PostgreSQL
 manual.][postgresql-hba].
 
-In this case, you want to give `new_user` in the `example.com` domain
-access to all databases on the data node, so add the following line to
-the `pg_hba.conf`.
+### Reload server configuration
+
+To reload the server configuration, you can use the following command
+on the data node:
 
 ```
-host    all    new_user    .example.com    scram-sha-256
+pg_ctl reload
 ```
 
-Once you have set up the authentication method on the data node, you
-need to check the configuration file (normally `postgresql.conf` in
-the data directory of the data node server) and make sure that the
-`hba_file` option is correctly set.
+### Add new user on data node
+
+You can add the new user on the data node by connecting to it using
+`psql` and using [`CREATE ROLE`][postgresql-create-role]:
+
+```
+CREATE ROLE new_user WITH LOGIN PASSWORD 'xyzzy';
+```
 
-### Setting up Passwords on the Access Node [](distdb-auth-pass-access-node)
+### Setting up Password Authentication on the Access Node [](distdb-auth-pass-access-node)
 
 To execute queries across data nodes, the access node needs to connect
 to data nodes as the user executing the query. This means that the
-access node needs passwords for the current user on each data node,
-and these are stored in a password file on the access node.
+access node needs to be configured for password authentication and
+need the passwords for the current user on each data node.
+
+To set up password authentication on the access node, you need to:
+
+1. Set the correct password encryption in the configuration file.
+2. Create a password file with passwords of the user on the data
+   nodes.
+3. Set password file to use in the configuration file.
+4. Reload the server configuration.
+
+### Set up password encryption method
```

Comment:
Also recommend increasing hierarchy here and the following two sections.

Reply:
Done.